src/usePreparedEffect.ts

```diff
@@ -0,0 +1,31 @@
+import { PrepareHookEffect, PrepareHookFunction } from './types';
+import { DependencyList, useEffect } from 'react';
+import { __REACT_PREPARE__ } from './constants';
+
+export interface PreparedEffectOptions {
+  runOnClient?: boolean;
+  awaitImmediately?: boolean;
+}
+
+const usePreparedEffect = (
+  prepareFunction: PrepareHookFunction,
+  deps: DependencyList = [],
+  opts: PreparedEffectOptions = {},
+) => {
+  const { runOnClient = true, awaitImmediately = false } = opts;
+
+  const effect = (): void => {
+    if (runOnClient) {
+      prepareFunction();
+    }
+  };
```

Comment:
Hmm, I feel like the whole `runOnClient` option is kinda unecessary here :thinking:.
Why not just follow the same api as `useEffect`, and just create the effect on the client if the deps is an array with > 0 elements? In the same fashion. I.e `runOnClient` can be inferred from `deps`
- `deps == []`: Only run on server render
- `deps == [foo, bar]`: Create `useEffect` with these options
- `deps == undefined` (should we support this at all?): Always run on update? I.e. create `useEffect` with no deps.

Reply:
I agree it's kind of unneccesary, but what you've described isn't really the same API as useEffect.
`deps == []` will run once on the client, and I think it would kind of annoying if it didn't as we have quite a few `prepare`-functions in `lego-webapp` that run before ssr and once on initial client load.

We could make `deps == undefined` never run on the client, but I think it's better to follow the behaiviour of `useEffect` exactly as it would be easier to understand for someone not familiar with the library.

We actually don't really need the option at all, it is never used in `lego-webapp`, but I also don't really see the harm, as it takes _two_ lines of code to implement.